x-pack/plugins/observability_solution/entity_manager/server/lib/entities/install_entity_definition.ts

```diff
@@ -218,3 +188,123 @@ async function installAndStartDefinition(
+    ...definition,
+    installStatus: 'installing',
+    installStartedAt: new Date().toISOString(),
+  });
+
+  logger.debug(`Installing index templates for definition ${definition.id}`);
+  await Promise.allSettled([
+    upsertTemplate({
+      esClient,
+      logger,
+      template: getEntitiesHistoryIndexTemplateConfig(definition),
+    }),
+    upsertTemplate({
+      esClient,
+      logger,
+      template: getEntitiesLatestIndexTemplateConfig(definition),
+    }),
+  ]).then(throwIfRejected);
+
+  logger.debug(`Installing ingest pipelines for definition ${definition.id}`);
+  await Promise.allSettled([
+    createAndInstallHistoryIngestPipeline(esClient, definition, logger),
+    createAndInstallLatestIngestPipeline(esClient, definition, logger),
+  ]).then(throwIfRejected);
+
+  logger.debug(`Installing transforms for definition ${definition.id}`);
+  await Promise.allSettled([
+    createAndInstallHistoryTransform(esClient, definition, logger),
+    isBackfillEnabled(definition)
+      ? createAndInstallHistoryBackfillTransform(esClient, definition, logger)
+      : Promise.resolve(),
+    createAndInstallLatestTransform(esClient, definition, logger),
+  ]).then(throwIfRejected);
+
+  await updateEntityDefinition(soClient, definition.id, { installStatus: 'installed' });
+
+  return entityDefinition;
+}
+
+// stop and delete the current transforms and reinstall all the components
+async function reinstall({
+  esClient,
+  soClient,
+  definition,
+  latestDefinition,
+  logger,
+}: InstallDefinitionParams & { latestDefinition: EntityDefinition }): Promise<EntityDefinition> {
+  logger.debug(
+    `Reinstalling definition ${definition.id} from v${definition.version} to v${latestDefinition.version}`
+  );
+
+  try {
+    await updateEntityDefinition(soClient, latestDefinition.id, {
+      ...latestDefinition,
+      installStatus: 'installing',
+      installStartedAt: new Date().toISOString(),
+    });
+
+    logger.debug(`Stopping transforms for definition ${definition.id} v${definition.version}`);
+    await Promise.all([
+      stopAndDeleteHistoryTransform(esClient, definition, logger),
+      isBackfillEnabled(definition)
+        ? stopAndDeleteHistoryBackfillTransform(esClient, definition, logger)
+        : Promise.resolve(),
+      stopAndDeleteLatestTransform(esClient, definition, logger),
+    ]);
+
+    await install({
+      esClient,
+      soClient,
+      logger,
+      definition: latestDefinition,
+    });
+
+    return latestDefinition;
+  } catch (err) {
+    await updateEntityDefinition(soClient, latestDefinition.id, {
+      installStatus: 'failed',
+    });
+
+    throw err;
+  }
+}
+
+const INSTALLATION_TIMEOUT = 5 * 60 * 1000;
+const shouldReinstall = (
+  definition: EntityDefinitionWithState,
+  latestDefinition: EntityDefinition
+) => {
+  const { installStatus, installStartedAt } = definition;
+
+  const isStale =
+    installStatus === 'installing' &&
```

Comment:
One important thing here is that these checks are not bwc since we assume the saved objects have `installStatus` persisted. With this in place the clusters with the builtin definitions stored won't be upgraded since they don't satisfy any of the condition. three solutions:
- migrate saved objects to set `updateStatus` and `updateStartedAt` to default values
- handle the case where no updateStatus exist here
- it's ok

Reply:
I'd add the saved object migrations, since we need to update the mappings of the saved objects anyway I think? Which would require us to add another migration as well.

Reply:
We don't need to add mappings unless we want to query by these fields which I don't think is that useful now

Reply:
Fixed in https://github.com/elastic/kibana/pull/188351/commits/d2545370b5b593605711fdeb172fba69b4332041